discovery-provider/src/tasks/index.py

```diff
@@ -254,13 +256,106 @@ def fetch_tx_receipts(self, block):
     return block_tx_with_receipts
 
 
-def fetch_ipfs_metadata(
+def get_metadata_from_json(default_metadata_fields, resp_json):
+    metadata = {}
+    for parameter, value in default_metadata_fields.items():
+        metadata[parameter] = (
+            resp_json.get(parameter) if resp_json.get(parameter) != None else value
+        )
+    return metadata
+
+
+async def fetch_metadata_from_gateway_endpoints(
```

Comment:
we definitely need docs for this function

Reply:
also any reason this has to be index.py vs another file?

Reply:
yeah maybe it can just be moved to the IPFSClient which should probably be renamed to content node client. eventually, the "/ipfs" route should probably be changed too.

Reply:
yeah we can definitely name this something else. not sure if you want to do that here or in a follow up PR

Reply:
created a follow up task for renaming route and client: https://linear.app/audius/issue/ASI-899/rename-ipfs-route-and-client.